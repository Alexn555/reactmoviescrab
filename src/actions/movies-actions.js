import { client } from './';

const apiKey = '67aeed398efbb2e1a3d33ea43395c16f';
const locale = 'en-US';

export function fetchItems(page = 1){
  return dispatch => {
    dispatch({
      type: 'FETCH_MOVIES',
      payload: client.get('3/movie/popular?api_key='+apiKey+'&language='+locale+'&page='+page)
    })
  }
}


