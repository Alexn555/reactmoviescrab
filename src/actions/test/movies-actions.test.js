import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import fetchMock from 'fetch-mock';
import { fetchItems } from '../movies-actions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

const fetchMovies = (page) => ({
  type: 'FETCH_MOVIES',
  page
});

describe('async actions', () => {
  afterEach(() => {
    fetchMock.restore()
  });
  
  it('should send FETCH_MOVIES action and get correct snapshot', () => {
	const store = mockStore({ movies: [], loading: [], errors: [] });
	const page = 1;
    store.dispatch(fetchMovies(page));
    expect(store.getActions()).toMatchSnapshot();
  });
  
  it('should send FETCH_MOVIES action and get right actions', () => {
    fetchMock.getOnce('/movies', {
      body: { payload: ['something'] },
      headers: { 'content-type': 'application/json' }
    });
	const page = 1;

    const expectedActions = [
      { type: 'FETCH_MOVIES', page: 1 } 
    ];
	
    const store = mockStore({ movies: [], loading: [], errors: [] });
	store.dispatch(fetchMovies(page));
    expect(store.getActions()).toEqual(expectedActions);
  });
  
  
})




