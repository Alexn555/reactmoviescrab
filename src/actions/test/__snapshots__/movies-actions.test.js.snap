// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async actions should send FETCH_MOVIES action and get correct snapshot 1`] = `
Array [
  Object {
    "page": 1,
    "type": "FETCH_MOVIES",
  },
]
`;
